src/client/eppo-client-with-bandits.spec.ts

```diff
@@ -40,16 +47,179 @@ describe('EppoClient Bandits E2E test', () => {
     const httpClient = new FetchHttpClient(apiEndpoints, 1000);
     const configurationRequestor = new ConfigurationRequestor(httpClient, flagStore, banditStore);
     await configurationRequestor.fetchAndStoreConfigurations();
+  });
 
+  beforeEach(() => {
     client = new EppoClient(flagStore, banditStore);
     client.setIsGracefulFailureMode(false);
+    client.setAssignmentLogger({ logAssignment: mockLogAssignment });
+    client.setBanditLogger({ logBanditAction: mockLogBanditAction });
+    jest.clearAllMocks();
   });
 
   afterAll(() => {
     jest.restoreAllMocks();
   });
 
-  it('TODO', () => {
-    // TODO: bandit test cases with the client
+  describe('Shared test cases', () => {
+    const testData = readBanditTestData();
+    // Build a map for more useful test names
+    const testsByFlagKey: Record<string, BanditTestCase> = {};
+    testData.forEach((testCase) => (testsByFlagKey[testCase.flag] = testCase));
+
+    it.each(Object.keys(testsByFlagKey))(
+      'Shared bandit test case - %s',
+      async (flagKey: string) => {
+        const { defaultValue, subjects } = testsByFlagKey[flagKey];
+        let numAssignmentsChecked = 0;
+        subjects.forEach((subject) => {
+          // TODO: handle already-bucketed attributes
+          // TODO: common test case with a numeric value passed as a categorical attribute and vice verse
```

Comment:
This will be done in subsequent pull requests. For now, I consider any attributes that are numbers numeric, and everything else categorical.